private/extensions/maven.bzl

```diff
@@ -0,0 +1,348 @@
+    # - fail_on_missing_checksum: bool. A logical OR over all `fail_on_missing_checksum` for all `install` tags with the same name.
+    # - fetch_javadoc: bool. A logical OR over all `fetch_javadoc` for all `install` tags with the same name.
+    # - fetch_sources: bool. A logical OR over all `fetch_sources` for all `install` tags with the same name.
+    # - jetify: bool. A logical OR over all `jetify` for all `install` tags with the same name.
+    # - jetify_include_list: string list. Accumulated from all `install` tags
+    # - lock_file: the lock file to use, if present. Multiple lock files will cause the build to fail.
+    # - overrides: a dict mapping `artfifactId:groupId` to Bazel label.
+    # - repositories: the list of repositories to pull files from.
+    # - strict_visibility: bool. A logical OR over all `strict_visibility` for all `install` tags with the same name.
+    # - strict_visibility_value: a string list. Build will fail is duplicated and different.
+    # - use_starlark_android_rules: bool. A logical OR over all `use_starlark_android_rules` for all `install` tags with the same name.
+    # - use_unsafe_shared_cache: bool. A logical OR over all `use_unsafe_shared_cache` for all `install` tags with the same name.
+    # - version_conflict_policy: string. Fails build if different and not a default.
+
+    for mod in mctx.modules:
+        for override in mod.tags.override:
+            value = str(override.target)
+            current = overrides.get(override.coordinates, None)
+            to_use = _fail_if_different("Target of override for %s" % override.coordinates, current, value, [None])
+            overrides.update({override.coordinates: to_use})
+
+        for artifact in mod.tags.artifact:
+            repo = repos.get(artifact.name, {})
+            existing_artifacts = repo.get("artifacts", [])
+
+            to_add = {
+                "group": artifact.group,
+                "artifact": artifact.artifact,
+            }
+
+            if artifact.version:
+                to_add.update({"version": artifact.version})
+
+            if artifact.packaging:
+                to_add.update({"packaging": artifact.packaging})
+
+            if artifact.classifier:
+                to_add.update({"classifier": artifact.classifier})
+
+            if artifact.neverlink:
+                to_add.update({"neverlink": artifact.neverlink})
+
+            if artifact.testonly:
+                to_add.update({"version": artifact.testonly})
+
+            if artifact.exclusions:
+                to_add.update({"exclusions": artifact.exclusions})
+
+            existing_artifacts.append(to_add)
+            repo["artifacts"] = existing_artifacts
+            repos[artifact.name] = repo
+
+        for install in mod.tags.install:
+            repo = repos.get(install.name, {})
+
+            artifacts = repo.get("artifacts", [])
+            repo["artifacts"] = artifacts + install.artifacts
```

Comment:
Does this mean that if module A depends on module B, and both A and B has `maven.install` that use the same `install.name`, their artifacts will be flattened into the same list?